crates/schema/src/composed_schema.rs

```diff
@@ -238,7 +245,20 @@ impl ComposedSchema {
         composed_schema.mutation_type = Some(Name::new("Mutation"));
         composed_schema.subscription_type = Some(Name::new("Subscription"));
 
+        // Store the original type definitions for validation later
+        let mut original_type_definitions = HashMap::new();
+
         for (service, doc) in federation_sdl {
+            for definition in doc.definitions.clone() {
```

Comment:
Maybe cloning here is to early and better to clone only `type_definition` in the `insert`?